packaging/build_rpm.py

```diff
@@ -336,22 +338,46 @@ def install_mock():
 
 
 def check_mock_config():
-    # allow network access during the build
-    # (we have to download packages from pypi)
-    mock_config = check_output(
-        MOCK + ['--debug-config',
-        ]).splitlines()
-
-    for line in mock_config:
-        if all(s in line for s in ('rpmbuild_networking', 'True')):
-            break
-    else:
-        logger.info('configuring rpmbuild_networking')
-        check_call(
-                '''echo "config_opts['rpmbuild_networking'] = True" '''
-                '| sudo -n tee -a /etc/mock/site-defaults.cfg',
-                shell=True,
-                )
+    """
+    allow network access during the build and set the %dist macro
+    (we have to download packages from pypi)
+    """
+    mock_config = check_output(MOCK + ['--debug-config', '-qq'])
+
+    config_opts = {}
+    config = {'config_opts': config_opts}
+    exec mock_config in config, config
+
+    options = [
+        "config_opts['rpmbuild_networking'] = True",
+        "config_opts['macros']['%dist'] = '.el7'",
+    ]
+    for option in reversed(options):
```

Comment:
Why not just change the order of options in the first place, instead of using `reversed`?

Reply:
The order isn't important, using reversed allows us to safely delete elements from the list we're iterating over